src/global-firestore-data/semesters.ts

```diff
@@ -104,7 +102,6 @@ export const addCourseToSemester = (
     if (semesterFound) return newSemestersWithCourse;
     return [...oldSemesters, createSemester(year, season, [newCourse])];
   });
-  updateRequirementChoice(newCourse.uniqueID, choiceUpdater);
```

Comment:
As mentioned before, I'm decoupling `addCourseToSemester` and `updateRequirementChoice`.